google-http-client/src/main/java/com/google/api/client/http/HttpResponse.java

```diff
@@ -356,6 +357,44 @@ public InputStream getContent() throws IOException {
     return content;
   }
 
+  static class ConsumingInputStream extends InputStream {
```

Comment:
An outer class would better separate unrelated code. This is complex enough to justify it.